exp/examine/local.go

```diff
@@ -0,0 +1,164 @@
+package examine
+
+import (
+	"os"
+	"strings"
+
+	"github.com/chanzuckerberg/fogg/errs"
+	"github.com/hashicorp/go-getter"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/spf13/afero"
+)
+
+const apiHostname = "https://registry.terraform.io"
+const apiVersion = "/v1"
+const resourceType = "/modules/"
+
+//**Local refers to any files located within your local file system**
+
+//GetLocalModules retrieves all terraform modules within a given directory
+//TODO:(EC) Define local and global modules OR rename the values
+func GetLocalModules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
+	_, err := os.Stat(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	modules, err := getAllModules(fs, dir)
+	if err != nil {
+		return nil, err
+	}
+	return modules, nil
+}
+
+//getAllModules retrieves all modules (and their respective resources, data, etc...) being used by the directory
+func getAllModules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
+	var queue []ModuleWrapper
+
+	//Find all submodules for the current directory
+	submodules, err := getSubmodules(fs, dir)
+	if err != nil {
+		return nil, err
+	}
+	queue = append(queue, submodules...)
+
+	//BFS on the current queue of submodules to find their submodules
+	for { //Do: recursively add new submodules to queue, While: queue is not empty
+
+		submods, err := getSubmodules(fs, queue[0].module.Path)
+		if err == nil {
+			//Add new elements to temp
+			queue = append(queue, submods...)
+			submodules = append(submodules, submods...)
+		}
+		//remove the current element
+		queue = append(queue[:0], queue[1:]...)
+
+		if !(len(queue) > 0) {
+			break
+		}
+	}
+
+	return submodules, nil
+}
+
+//getSubmodules retrieves all submodules related to the root directory
+func getSubmodules(fs afero.Fs, dir string) ([]ModuleWrapper, error) {
+	_, err := os.Stat(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	//FIXME: Make contains check module path so that repeats are not added
+	// if(modules != nil && contains(modules, dir))
+	var modules []ModuleWrapper
+	mod, diag := tfconfig.LoadModule(dir)
+	if diag.HasErrors() {
+		return nil, errs.WrapInternal(diag.Err(), "There was an issue loading the module")
+	}
+
+	//Loads all modules that the current module depends on
+	sources := getSources(mod)
+	for i, source := range sources {
+		if strings.HasPrefix(source.moduleSource, githubURL) { // If the module is a github link, specifically cztack
+			mod, err := GetFromGithub(fs, source.moduleSource)
+			if err != nil {
+				return nil, err
+			}
+
+			sources[i].module = mod
+			sources[i].version = getVersion(sources[i])
+			modules = append(modules, sources[i])
+		} else if strings.HasPrefix(source.moduleSource, awsRegistry) { // If the module is from the tf registry
```

Comment:
I don't think we need to download modules. It's not 100% clear yet but it looks like you're trying to recursively find all modules. If that's the case we can simplify things– since we can't update nested module sources, there is no need to fetch them and try to calculate updates.

Reply:
I guess my thought was that those nested modules might give us more information on what to update.